### PR TITLE
Refactor SPELL_REQUIRES_OFF_CENTCOM to SPELL_REQUIRES_STATION

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

When I refactored stuff to use lazy templates it broke a few needed checks, this is one of those checks.
Although there is no reason this shouldn't have been checking for station level over not CC
## Why It's Good For The Game

Resolves #73320 
## Changelog
:cl:
fix: Spells which are only able to be cast on the station can now only be cast on the station
/:cl:

### CODE REVIEW

File: code/__DEFINES/magic.dm [hunk 1]
```diff
@@ -55,7 +55,7 @@
 /// Whether the spell can be cast while the user has antimagic on them that corresponds to the spell's own antimagic flags.
 #define SPELL_REQUIRES_NO_ANTIMAGIC (1 << 4)
 /// Whether the spell can be cast on the centcom z level.
```
Comment (Fikou) [inline code/__DEFINES/magic.dm:57]: update it

Comment (ZephyrTFA) [reply code/__DEFINES/magic.dm:57]: ah fuck

**Comment by itseasytosee** [comment]
Why wouldn't it be valid to cast these spells in non-station non-centcomm space z-levels?

**Comment by tralezab** [comment]
> Why wouldn't it be valid to cast these spells in non-station non-centcomm space z-levels?

Because the wizard ship is on a non-station non-centcomm space z-level. Did ya read the issue?

**Comment by tralezab** [comment]
And as a side note, the wizard should not really be anywhere that isn't the den or the station in the first place.

**Comment by necromanceranne** [comment]
> And as a side note, the wizard should not really be anywhere that isn't the den or the station in the first place.

Shouldn't =/= can't. And just drifting off station z-level isn't a tall order. That said, how many spells does this actually affect?

**Comment by ZephyrTFA** [comment]
> > And as a side note, the wizard should not really be anywhere that isn't the den or the station in the first place.
> 
> Shouldn't =/= can't. And just drifting off station z-level isn't a tall order. That said, how many spells does this actually affect?

5

**Review by MrMelbert — APPROVED** [review]

**Pull request merged**